File: src/sidecar/capture/ice_servers.py
#!/usr/bin/env python3
"""
Centralized ICE Server Configuration for Python backends.
Single source of truth for all STUN/TURN servers.
Mirrors src/scripts/core/network/ice-servers.js
"""

# STUN Servers — kept to 3 (mirrors ice-servers.js). More than ~4 slows
# ICE discovery on every offer. Python format (stun://)
STUN_SERVERS = [
    "stun://stun.l.google.com:19302",
    "stun://stun1.l.google.com:19302",
    "stun://stun.cloudflare.com:3478",
]

# JavaScript format (stun:) for reference
STUN_SERVERS_JS = [
    "stun:stun.l.google.com:19302",
    "stun:stun1.l.google.com:19302",
    "stun:stun.cloudflare.com:3478",
]


def get_stun_servers():
    """Return list of STUN servers in Python format."""
    return STUN_SERVERS


def get_stun_servers_js():
    """Return list of STUN servers in JavaScript format."""
    return STUN_SERVERS_JS


def build_ice_servers(turn_servers=None):
    """
    Build ICE server configuration for webrtcbin.
    
    Args:
        turn_servers: List of TURN server dicts with keys: urls, username, credential
    
    Returns:
        dict with 'stun' (list) and 'turn' (list) keys
    """
    return {
        'stun': STUN_SERVERS,
        'turn': turn_servers or [],
    }


if __name__ == '__main__':
    import json
    print(json.dumps({
        'stun': STUN_SERVERS,
        'stun_js': STUN_SERVERS_JS,
    }, indent=2))

File: src/sidecar/capture/gst-nearcade/src/ice_servers.rs
//! Centralized ICE Server Configuration for Rust backends.
//! Mirrors src/scripts/core/network/ice-servers.js and src/sidecar/capture/ice_servers.py

/// STUN Servers — kept to 3 (mirrors ice-servers.js). More than ~4 slows
/// ICE discovery on every offer; the dead public TURNs stay out entirely
/// (real TURN comes from server /api/turn, not hardcoded).
/// Python/Rust format (stun://)
pub const STUN_SERVERS: &[&str] = &[
    "stun://stun.l.google.com:19302",
    "stun://stun1.l.google.com:19302",
    "stun://stun.cloudflare.com:3478",
];

/// JavaScript format (stun:) for reference
pub const STUN_SERVERS_JS: &[&str] = &[
    "stun:stun.l.google.com:19302",
    "stun:stun1.l.google.com:19302",
    "stun:stun.cloudflare.com:3478",
];

/// Get STUN servers in Rust format
pub fn stun_servers() -> &'static [&'static str] {
    STUN_SERVERS
}

/// Get STUN servers in JavaScript format
pub fn stun_servers_js() -> &'static [&'static str] {
    STUN_SERVERS_JS
}

/// Build ICE server configuration for webrtcbin
pub fn build_ice_servers(turn_servers: Option<Vec<TurnServer>>) -> IceServers {
    IceServers {
        stun: STUN_SERVERS.iter().map(|s| s.to_string()).collect(),
        turn: turn_servers.unwrap_or_default(),
    }
}

#[derive(Debug, Clone)]
pub struct TurnServer {
    pub urls: Vec<String>,
    pub username: String,
    pub credential: String,
}

#[derive(Debug, Clone)]
pub struct IceServers {
    pub stun: Vec<String>,
    pub turn: Vec<TurnServer>,
}

impl Default for IceServers {
    fn default() -> Self {
        Self {
            stun: STUN_SERVERS.iter().map(|s| s.to_string()).collect(),
            turn: vec![],
        }
    }
}